### PR TITLE
[ci]: skipping debug builds when not making a release

### PR DESCRIPTION
**Pull request by psk001**
Adjusted the build scripts to skip building debug versions unless the GH actions runs for a release by including a SKIP_DEBUG variable in command.

### CODE REVIEW

**Comment by psk001** [comment]
@yurishkuro please review the PR and suggest changes.

File: scripts/build-all-in-one-image.sh [hunk 1]
```diff
@@ -46,10 +48,14 @@ run_integration_test localhost:5000/$repo
...
+if [ "$SKIP_DEBUG" = true ] ; then
+  make build-all-in-one-debug GOOS=linux GOARCH=$GOARCH
+  repo=${repo}-debug
+  #build all-in-one-debug image locally for integration test
+  bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one-debug -d cmd/all-in-one -t debug
+  run_integration_test localhost:5000/$repo
+  #build all-in-one-debug image and upload to dockerhub/quay.io
+  bash scripts/build-upload-a-docker-image.sh -b -c all-in-one-debug -d cmd/all-in-one -t debug
+else 
+  echo "Skipping debug commands"
```
Comment (yurishkuro) [inline scripts/build-all-in-one-image.sh:60]: looks like this should be in the `then` part

File: .github/workflows/ci-all-in-one-build.yml [hunk 2]
```diff
@@ -48,3 +48,4 @@ jobs:
       env:
         DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
         QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+        SKIP_DEBUG: ${{inputs.SKIP_DEBUG}}
```
Comment (yurishkuro) [inline .github/workflows/ci-all-in-one-build.yml:51]: We don't run actions manually. The main release is done by `.github/workflows/ci-release.yml`, so there it should be enabling debug builds, and here they should be explicitly disabled.

**Review by yurishkuro — COMMENTED** [review]
Another place where debug images are built is scripts/build-upload-docker-images.sh

Note that that script already accepts `pr-only` argument, so we can adjust it directly to exclude debug logic for pr-only, without any changes to the workflows. And for consistency we should implement a similar flag for `scripts/build-all-in-one-image.sh`, instead of the new SKIP_DEBUG variable.

**Comment by psk001** [comment]
Got them all, working on it

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**